File: solutions/go/difference-of-squares/1/difference_of_squares.go
package diffsquares

func SquareOfSum(n int) (result int) {
	for i := 0; i <= n; i++ {
		result += i
	}

	return result * result
}

func SumOfSquares(n int) (result int) {
	for i := 0; i <= n; i++ {
		result += i * i
	}

	return
}

func Difference(n int) int {
	return SquareOfSum(n) - SumOfSquares(n)
}

File: solutions/typescript/difference-of-squares/1/difference-of-squares.ts
export class Squares {
  constructor(private readonly count: number) {}

  get sumOfSquares(): number {
    return this.range(1, this.count).reduce((acc, n) => acc + n ** 2, 0)
  }

  get squareOfSum(): number {
    return this.range(1, this.count).reduce((acc, n) => acc + n) ** 2
  }

  get difference(): number {
    return this.squareOfSum - this.sumOfSquares
  }

  private *range(start: number, end: number) {
    for (let i = start; i <= end; i++) yield i
  }
}
